### PR TITLE
docs: remove prerequisite section in NgModules docs

### PR DESCRIPTION
**Pull request by kapunahelewong**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Thereʻs a section at the top of each NgModules page with a list of prerequisites. There have been several issues where readers have been confused by them (either because the nav was not in the same order or because they werenʻt sequential). We also havenʻt continued this formatting in other rewrites of the docs. 

Issue Number: #21531

## What is the new behavior?
Remove this section altogether. It will shorted each document, clear clutter on the page, and allow the reader to find the pertinent information more quickly.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/ngmodule-vs-jsmodule.md [hunk 1]
```diff
@@ -1,10 +1,5 @@
 # JavaScript Modules vs. NgModules
 
-#### Prerequisites
```
Comment (brandonroberts) [inline aio/content/guide/ngmodule-vs-jsmodule.md]: I think we should still link to the JavaScript modules page.

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: aio/content/guide/ngmodule-vs-jsmodule.md [hunk 2]
```diff
@@ -1,10 +1,5 @@
...
-A basic understanding of [JavaScript/ECMAScript modules](https://hacks.mozilla.org/2015/08/es6-in-depth-modules/).
-
-<hr>
-
 JavaScript and Angular use modules to organize code, and
 though they organize it differently, Angular apps rely on both.
```
Comment (brandonroberts) [inline aio/content/guide/ngmodule-vs-jsmodule.md]: Will you link `modules` in the text below to https://hacks.mozilla.org/2015/08/es6-in-depth-modules/ so we don't lose the reference material.

**Review by brandonroberts — DISMISSED** [review]

File: aio/content/guide/ngmodule-vs-jsmodule.md (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/guide/ngmodule-vs-jsmodule.md]: Done!

File: aio/content/guide/ngmodule-vs-jsmodule.md (hunk 2 above)
Comment (kapunahelewong) [reply aio/content/guide/ngmodule-vs-jsmodule.md]: Done!

**Review by brandonroberts — COMMENTED** [review]
LGTM. I'll leave final approval to @IgorMinar

**Comment by kapunahelewong** [comment]
Aw, thanks @brandonroberts!!

**Review by IgorMinar — APPROVED** [review]

**Comment by IgorMinar** [comment]
merge-assistance: global approval

**Pull request closed without merging**